src/scm/gitlab.js

```diff
@@ -0,0 +1,278 @@
+          context.repo = repo;
+        }
+        return repos;
+      });
+  }
+
+  createRepo() {
+    const owner = $('#new-repo-owner').val();
+    const name = $('#new-repo-name').val();
+    const desc = $('#new-repo-desc').val();
+    const visibility = ($('#new-repo-type').val() !== 'public') ? 'private' : 'public';
+    const payload = {
+      name: name,
+      description: desc,
+      visibility: visibility
+    };
+    if (!name || name === '') return;
+    return new Promise((resolve, reject) => {
+      return $.ajax({
+        url: `${this.baseUrl}/projects`,
+        headers: this.tokenHeader,
+        method: 'POST',
+        crossDomain: true,
+        dataType: 'json',
+        contentType: 'application/json',
+        data: JSON.stringify(payload)
+      })
+        .then(resolve)
+        .fail(reject);
+    })
+      .then(response => {
+        const repo = {
+          fullName: response.name,
+          id: response.id
+        };
+        context.repo = repo;
+        Object.assign(context.bindRepo, {[context.id]: repo});
+        if (context.bindBranch[context.id]) {
+          delete context.bindBranch[context.id];
+        }
+        chrome.storage.sync.set({bindRepo: context.bindRepo});
+        return response.name;
+      })
+      .then(repo => {
+        return this.commitFiles(repo, 'master', null, [{name: "README.md", content: "initialed by gas-github"}], null,null, 'initial commit')
+          .then(() => {
+            return repo;
+          });
+      })
+      .catch((err) => {
+        throw new Error('Failed to create new repository.');
+      });
+  }
+
+  createBranch() {
+    const branch = $('#new-branch-name').val();
+    const payload = {
+      branch: branch,
+      ref: context.branch
+    };
+    if (!branch || branch === '') return;
+    return new Promise((resolve, reject) => {
+        return $.ajax({
+          url: `${this.baseUrl}/projects/${context.repo.id}/repository/branches`,
+          headers: this.tokenHeader,
+          method: 'POST',
+          crossDomain: true,
+          dataType: 'json',
+          contentType: 'application/json',
+          data: JSON.stringify(payload)
+        })
+          .then(resolve)
+          .fail(reject);
+      })
+        .then(response => {
+          context.branch = branch;
+          Object.assign(context.bindBranch, { [context.id] : branch });
+          chrome.storage.sync.set({ bindBranch: context.bindBranch });
+          return branch;
+        })
+      .catch(err => {
+        throw new Error('Failed to create new branch.');
+      });
+  }
+
+  followPaginate(data) {
+    return new Promise((resolve, reject) => {
+      $.getJSON(data.url)
+        .then((response, status, xhr) => {
+          data.items = data.items.concat(response);
+          const link = xhr.getResponseHeader('Link');
+          let url = null;
+          if (link) {
+            const match = link.match(/<(.*?)>; rel="next"/);
```

Comment:
Lazy match will try to match as least as possible **after** the start point.
So currently this will match:


```
https://gitlab.com/api/v4/groups?+rel=%22prev%22%2C+%3Chttps%3A%2F%2Fgitlab.com%2Fapi%2Fv4%2Fgroups%3F0%3D&0=&order_by=name&owned=false&page=2&per_page=20&sort=asc&statistics=false&with_custom_attributes=false>; rel="prev", <https://gitlab.com/api/v4/groups?+rel=%22prev%22%2C+%3Chttps%3A%2F%2Fgitlab.com%2Fapi%2Fv4%2Fgroups%3F0%3D&0=&order_by=name&owned=false&page=4&per_page=20&sort=asc&statistics=false&with_custom_attributes=false
```
from 
```
"<https://gitlab.com/api/v4/groups?+rel=%22prev%22%2C+%3Chttps%3A%2F%2Fgitlab.com%2Fapi%2Fv4%2Fgroups%3F0%3D&0=&order_by=name&owned=false&page=2&per_page=20&sort=asc&statistics=false&with_custom_attributes=false>; rel="prev", <https://gitlab.com/api/v4/groups?+rel=%22prev%22%2C+%3Chttps%3A%2F%2Fgitlab.com%2Fapi%2Fv4%2Fgroups%3F0%3D&0=&order_by=name&owned=false&page=4&per_page=20&sort=asc&statistics=false&with_custom_attributes=false>; rel="next", <https://gitlab.com/api/v4/groups?+rel=%22prev%22%2C+%3Chttps%3A%2F%2Fgitlab.com%2Fapi%2Fv4%2Fgroups%3F0%3D&0=&order_by=name&owned=false&page=1&per_page=20&sort=asc&statistics=false&with_custom_attributes=false>; rel="first", <https://gitlab.com/api/v4/groups?+rel=%22prev%22%2C+%3Chttps%3A%2F%2Fgitlab.com%2Fapi%2Fv4%2Fgroups%3F0%3D&0=&order_by=name&owned=false&page=5231&per_page=20&sort=asc&statistics=false&with_custom_attributes=false>; rel="last""
```

please change to something like `<([^ ]*?)>; rel="next"`